mantidproject/mantid
# load cansas data without idev

Fixes #17002

This PR fixes an issue with LoadCanSAS which required Idev to be mandatory. The CanSAS standard states that the Idev entry is optional. See [here](http://www.cansas.org/formats/canSAS1d/1.1/doc/element_SASdata.html). 

This PR fixes this crash. If the Idev field is missing, then the strategy is to use the square root of the intensity as a measure of the error. If the Idev field is present, then just use it.
# To Test

The test files are be found here:
[canSASTestFiles.zip](https://github.com/mantidproject/mantid/files/381031/canSASTestFiles.zip)

__ Load file with missing Idev information__
- Load _withoutIdev.xml_ 
  - Confirm that there is no error

**Load file with Idev information**
- Load _withIdev.xml_ 
  - Confirm that there is no error
# Release Nots

See [here](https://github.com/mantidproject/mantid/pull/17005/commits/162f477609001dcd8dbbc47189cfc602f81950f3)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
